src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/ILinuxUtilizationParser.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Linux;
+
+/// <summary>
+/// An interface to be implemented by a parser that reads files and extracts resource utilization data from them.
+/// For both versions of cgroup controllers, the parser reads files from the /sys/fs/cgroup directory.
+/// </summary>
+internal interface ILinuxUtilizationParser
+{
+    /// <summary>
+    /// Reads file /sys/fs/cgroup/memory.max, which  is used to check the maximum amount of memory that can be used by a cgroup.
+    /// It is part of the cgroup v2 memory controller.
+    /// </summary>
+    /// <returns>maybeMemory.</returns>
+    ulong GetAvailableMemoryInBytes();
+
+    /// <summary>
+    /// Reads the file /sys/fs/cgroup/cpu.stat, which is part of the cgroup v2 CPU controller.
+    /// It provides statistics about the CPU usage of a cgroup.
+    /// </summary>
+    /// <returns>nanoseconds.</returns>
+    long GetCgroupCpuUsageInNanoseconds();
```

Comment:
Do we need to use `Cgroup` in method names here?

Reply:
This method is specifically used to get metrics created by control group (cgroup). Another method GetHostCpuUsageInNanoseconds() reads /proc/cpuinfo which is for host in general.

Reply:
I see, thanks